main.py

```diff
@@ -0,0 +1,100 @@
+import ee
+import argparse
+import geemap
+from geopy.geocoders import Nominatim
+from geopy.exc import GeopyError
+from decouple import config
+
+HEAT_MAP_GEE_PROJECT = config("GEE_PROJECT")
+ee.Initialize(project=HEAT_MAP_GEE_PROJECT)
+
+def get_city_coordinates()->list:
+    """Asks user for city"""
+    try: 
+        city_name = str(input("Enter the city urban center you want to view: "))
+        geolocator = Nominatim(user_agent="urban_heat_island")
+
+        location = geolocator.geocode(city_name)
+        if location:
+            city_coordinates = [location.latitude, location.longitude]
+            print(f"The center coordinates for your city are {city_coordinates}")
+            return city_coordinates
+        else:
+            raise ValueError(f"Location for '{city_name}' not found.")
+    
+    except GeopyError as e:
+        raise ValueError(f"Geolocation service error: {e}")
+
+def main(coordinates):
+    print("Hello from urban-heat-island!")
+    ## Approximate bounding box for Austin, Texas, creating a layer with legal boundraies 
+    city = ee.Geometry.Point(coordinates)
+    table = ee.FeatureCollection("FAO/GAUL/2015/level2")
+    roi = table.filterBounds(city).map(lambda vec: vec.simplify(1500)) # Region of Interest
+
+    Map = geemap.Map()
+    Map.centerObject(roi, 10)
+
+    # Creates image collection of Landsat 8 images, during summer months, filtering out cloud coverage of at least 30%,
+    # To convert raw thermal data (from Band 10) to brightness temperature:
+    landsat = ee.ImageCollection("LANDSAT/LC08/C02/T1_L2").select('ST_B10')
+
+    filters = (
+        landsat.filterBounds(roi).filterDate('2020-01-01', '2020-12-31').filter(ee.Filter.lt('CLOUD_COVER', 40))
+        .map(lambda img: img.multiply(ee.Number(img.get("TEMPERATURE_MULT_BAND_ST_B10")))  # Gain
+            .add(ee.Number(img.get("TEMPERATURE_ADD_BAND_ST_B10")))  # Offset
+            .copyProperties(img, img.propertyNames())
+        )
+    )
+
+    num_images = filters.size().getInfo()
+    print(f'Number of images after filtering: {num_images}')
+
+    thermal_infra_image = landsat.median()
+    Map.addLayer(thermal_infra_image.clip(roi),{},'tir_median', False)
+
+    # Calculate the mean temprature value over a region (roi), setting the spatial resolution in meters (scale=100)
```

Comment:
Its more to keep track of what each section does as I learn the google earth engine and its contents. As well as any calculations. I will probably remove them but for now they help me know where what is.

Reply:
I see what you mean, I could technically try to make each layer into a sub function but I don't know if that would exactly lead to easier to read code. At the moment it flows in order of how the calculations are made, but if I see a need to separate some of the functionalities because they could be reused elsewhere then I can break it down then.

Reply:
Yeah good point: a. you're still prototyping / getting it to work, b. longer is not always bad if components logically fit together.